pages/docs/reference/whatsnew12.md

```diff
@@ -0,0 +1,458 @@
+`findViewById` calls in Android API level 26:
+
+```
+val button = findViewById(R.id.button) as Button
+```
+
+### Smart cast improvements
+
+When a variable is assigned from a safe call expression and checked for null, the smart cast is now applied to 
+the safe call receiver as well:
+
+<div class="sample" markdown="1" data-min-compiler-version="1.1">
+
+```kotlin
+fun countFirst(s: Any): Int {
+    //sampleStart
+    val firstChar = (s as? CharSequence)?.firstOrNull()
+    if (firstChar != null)
+       return s.count { it == firstChar } // s: Any is smart cast to CharSequence
+    
+    val firstItem = (s as? Iterable<*>)?.firstOrNull()
+    if (firstItem != null)
+       return s.count { it == firstItem } // s: Any is smart cast to Iterable<*>
+    //sampleEnd
+    
+    return -1
+}
+
+fun main(args: Array<String>) {
+    val string = "abacaba"
+    val countInString = countFirst(string)
+    println("called on \"$string\": $countInString")
+    
+    val list = listOf(1, 2, 3, 1, 2)
+    val countInList = countFirst(list)
+    println("called on $list: $countInList")
+}
+```
+</div>
+
+Also, smart casts in a lambda are now allowed for local variables that are only modified before the lambda:
+
+<div class="sample" markdown="1" data-min-compiler-version="1.1">
+
+```kotlin
+fun main(args: Array<String>) {
+    val flag = args.size == 0
+    
+    //sampleStart
+    var x: String? = null
+    if (flag) x = "Yahoo!"
+
+    run {
+        if (x != null) {
+            println(x.length) // x is smart cast to String
+        }
+    }
+    //sampleEnd
+}
+```
+</div>
+
+### Support for  ::foo as a shorthand for this::foo
+
+A bound callable reference to a member of `this` can now be written without explicit receiver, `::foo` instead 
+of `this::foo`. This also makes callable references more convenient to use in lambdas where you refer to a member 
+of the outer receiver.
+
+### Deprecation: data classes overriding copy
+
+When a data class inherited from a type that already had the `copy` function with default parameters, the 
+`copy` implementation generated for the data class used the defaults from the supertype, leading to counter-intuitive 
+behavior (or a runtime failure in a similar case when there were no default parameters in the supertype). 
+This has become deprecated with a warning in Kotlin 1.2 and will be an error in Kotlin 1.3.
+
+### Deprecation: Nested types in enum entries
+
+Inside enum entries, defining a nested type that is not an `inner class` has been deprecated due to issues in the 
+initialization logic. This causes a warning in Kotlin 1.2 and will become an error in Kotlin 1.3.
+
+### Deprecation: single named argument for vararg
+
+For consistency with array literals in annotations, passing a single item for a vararg parameter in the named 
+form (`foo(items = i)`) has been deprecated. Please use the spread operator with the corresponding 
+array factory functions:
+
+```
+foo(items = *intArrayOf(1))
```

Comment:
I think it looks a bit strange to recommend to use `intArrayOf` on the same page where we advertise the much prettier array literal syntax

Reply:
The literal syntax is not yet supported outside annotations.